contracts/Core/BlockManager.sol

```diff
@@ -256,27 +256,37 @@ contract BlockManager is Initializable, ACL, BlockStorage, StateManager, IBlockM
             return;
         }
 
-        uint8 pushAt = uint8(sortedProposedBlockIds[epoch].length);
-        for (uint8 i = 0; i < sortedProposedBlockIds[epoch].length; i++) {
-            if (proposedBlocks[epoch][sortedProposedBlockIds[epoch][i]].biggestInfluence < biggestInfluence) {
-                pushAt = i;
+        for (uint8 i = 0; i <= sortedProposedBlockIds[epoch].length; i++) {
+            // Push at last, have worst itr
+            if (i == sortedProposedBlockIds[epoch].length && sortedProposedBlockIds[epoch].length < parameters.maxAltBlocks()) {
+                sortedProposedBlockIds[epoch].push(blockId);
```

Comment:
if biggest influencer is incorrect, attacker can escape the check here.

Reply:
Please simplify logic if possible. Not intuitive and hard to debug.

Reply:
hmm sure 
here attacker cant escape 
as this condition is checked at the end of loop

so in between if its found that its incorrect biggestInfluence it reverts

Reply:
true. but gas intensive to compare biggestinfluence with every block.
https://github.com/razor-network/contracts/pull/483/files#diff-ef39b70abd99dba15254b650eeab6fe94780b7804fbd0715169e10e8ad67e198R271

Unnecessary if proposer can be slashed in dispute by proving higher biggestInfluence.
Which is missing and important.

Reply:
hmm yep 
okay cool will remove it, even if block with smaller influcence is passed, it will be included as of now.
Then in solution of #474, we will add it.

Reply:
Fixed https://github.com/razor-network/contracts/pull/483/commits/226903825512b9087390fa4d297389f3502cccdb